src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,56 +3,78 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
-    private boolean isPrivate;
-
-    /**
-     * Validates given address.
-     *
-     * @throws IllegalValueException if given address string is invalid.
-     */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
-    }
-
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
-    }
+	public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+	public static final String MESSAGE_ADDRESS_CONSTRAINTS = "a/BLOCK, STREET, UNIT, POSTAL_CODE";
+	public static final String ADDRESS_VALIDATION_REGEX = ".+";
+
+	public final String value;
+	private boolean isPrivate;
+	public Block block;
+	public Street street;
+	public Unit unit;
+	public Postalcode postalcode;
+
+	/**
+	 * Validates given address.
+	 *
+	 * @throws IllegalValueException
+	 *             if given address string is invalid.
+	 */
+	public Address(String address, boolean isPrivate) throws IllegalValueException {
+		this.isPrivate = isPrivate;
+		if (!isValidAddress(address)) {
+			throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+		}
+		this.value = address;
+		splitAddress();
+	}
+
+	/**
+	 * Returns true if a given string is a valid person email.
+	 */
+	public static boolean isValidAddress(String test) {
+		return test.matches(ADDRESS_VALIDATION_REGEX);
+	}
+
+	public void splitAddress() {
+		String[] subclasses = new String[4];
+		int i = 0;
+		for (String subAddress : value.split(",")) {
+			subclasses[i] = subAddress;
+			subclasses[i] = subclasses[i].trim();
+			++i;
+		}
+		block = new Block(subclasses[0]);
+		street = new Street(subclasses[1]);
+		unit = new Unit(subclasses[2]);
+		postalcode = new Postalcode(subclasses[3]);
+
+	}
+
+	@Override
+	public String toString() {
+		return block.getBlock() + " " + street.getStreet() + " " + unit.getUnit() + " " + postalcode.getPostalcode();
```

Comment:
`" "` isn't exactly a magic number...it's being repeated quite a bit. how about creating a constant variable for it?
e.g. `private static final WHITESPACE = " ";`